File: TopInterview150/Intervals/SummaryRanges/solution.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        int start = 0;
        vector<string> output;
        if (nums.size() == 0)
            return output;
        for (int i = 0; i < nums.size()-1; i++)
        {
            if (nums[i] + 1 != nums[i+1])//if next isn't conescutive
            {
                if (nums[start] == nums[i])
                {
                    output.push_back(to_string(nums[start]));
                }
                else
                {
                    output.push_back(to_string(nums[start]) + "->" + to_string(nums[i]));
                }
                start = i+1;
            }
        }
        //check last element
        if (nums[start] == nums[nums.size()-1])
        {
            output.push_back(to_string(nums[start]));
        }
        else
        {
            output.push_back(to_string(nums[start]) + "->" + to_string(nums[nums.size()-1]));
        }
        return output;
}
};

File: TopInterview150/Intervals/SummaryRanges/solution.py
class Solution(object):
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        start = 0
        output = []
        if len(nums) == 0:
            return output
        for i in range(len(nums)-1):
            if nums[i+1] != nums[i]+1:
                """if next num is not consecutive"""
                if start == i:
                    output.append(str(nums[i]))
                else:
                    output.append(str(nums[start]) + "->" + str(nums[i]))
                
                start = i+1  
        if nums[start] == nums[len(nums)-1]:
            output.append(str(nums[start]))
        else:
            output.append(str(nums[start]) + "->" + str(nums[len(nums)-1]))
        return output;  